hstest/testing/plotting/matplotlib_handler.py

```diff
@@ -0,0 +1,235 @@
+
+
+class MatplotlibHandler:
+    _saved = False
+    _replaced = False
+
+    _Axes = None
+    _hist = None
+    _plot = None
+    _scatter = None
+    _pie = None
+    _bar = None
+    _barh = None
+    _violinplot = None
+    _imshow = None
+    _boxplot = None
+    _show = None
+    _backend = None
+    _matplotlib = None
+
+    @staticmethod
+    def replace_plots(drawings):
+
+        try:
+            import matplotlib
+        except ModuleNotFoundError:
+            return
+
+        lib_type = 'matplotlib'
+
+        def custom_show_func(*args, **kwargs):
+            pass
+
+        def hist(x, *a, **kw):
+            drawing = Drawing(
+                lib_type,
+                'hist',
+                {
+                    'x': x
+                }
+            )
+            drawings.append(drawing)
+
+        def plot(*args, **kwargs):
+            drawing = Drawing(
+                lib_type,
+                'plot',
+                {
+                    'args': args
+                }
+            )
+            drawings.append(drawing)
+
+        def scatter(x, y, *a, **kwargs):
+            drawing = Drawing(
+                lib_type,
+                'scatter',
+                {
+                    'x': x,
+                    'y': y
+                }
+            )
+            drawings.append(drawing)
+
+        def pie(x, *a, **kw):
+            drawing = Drawing(
+                lib_type,
+                'pie',
+                {
+                    'x': x
+                }
+            )
+            drawings.append(drawing)
+
+        def bar(x, height, width=0.8, bottom=None, **kwargs):
+            drawing = Drawing(
+                lib_type,
+                'bar',
+                {
+                    'x': x,
+                    'height': height,
+                    'kwargs': kwargs
+                }
+            )
+            drawings.append(drawing)
+
+        def barh(y, width, height=0.8, left=None, **kwargs):
+            drawing = Drawing(
+                lib_type,
+                'barh',
+                {
+                    'y': y,
+                    'width': width,
+                    'kwargs': kwargs
+                }
+            )
+            drawings.append(drawing)
+
+        def violinplot(dataset, **kwargs):
+            drawing = Drawing(
+                lib_type,
+                'violin',
+                {
+                    'dataset': dataset,
+                    'kwargs': kwargs
+                }
+            )
+            drawings.append(drawing)
+
+        def imshow(x, **kwargs):
+            drawing = Drawing(
+                lib_type,
+                'heatmap',
+                {
+                    'x': x,
+                    'kwargs': kwargs
+                }
+            )
+            drawings.append(drawing)
+
+        def boxplot(x, **kwargs):
+            drawing = Drawing(
+                lib_type,
+                'boxplot',
```

Comment:
Why `violinplot` type is `violin` but `boxplot` type is `boxplot`? I think we should remove `plot` in every drawing type, just leave `violin` and `box`.